{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "STfdGh5Ui6Tn"
   },
   "source": [
    "#How To Train YOLOv5 Oriented Bounding Boxes Custom Object Detection\n",
    "\n",
    "Oriented bounding boxes are bounding boxes rotated to better fit the objects represented on an angle. In this notebook, we use the YOLOv5-OBB repository in order to train an already annotated dataset to view the Oriented bounding boxes detection at play.\n",
    "\n",
    "\n",
    "## Instructions\n",
    "\n",
    "Run one cell at at time and run:\n",
    "\n",
    "```\n",
    ">>> import torch\n",
    ">>> torch.version.cuda\n",
    ">>> exit()\n",
    "```\n",
    "\n",
    "when prompted to in the first cell. \n",
    "\n",
    "Addidtional Resources:\n",
    "\n",
    "- [Yolov5-OBB installation instructions](https://github.com/hukaixuan19970627/yolov5_obb/blob/master/docs/install.md)\n",
    "- [Getting started](https://github.com/hukaixuan19970627/yolov5_obb/blob/master/docs/GetStart.md)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Get:1 http://archive.ubuntu.com/ubuntu focal InRelease [265 kB]\n",
      "Get:2 http://security.ubuntu.com/ubuntu focal-security InRelease [114 kB]\n",
      "Get:3 http://security.ubuntu.com/ubuntu focal-security/multiverse amd64 Packages [29.3 kB]\n",
      "Get:4 http://archive.ubuntu.com/ubuntu focal-updates InRelease [114 kB]\n",
      "Get:5 http://security.ubuntu.com/ubuntu focal-security/universe amd64 Packages [1125 kB]\n",
      "Get:6 http://archive.ubuntu.com/ubuntu focal-backports InRelease [108 kB]\n",
      "Get:7 http://archive.ubuntu.com/ubuntu focal/multiverse amd64 Packages [177 kB]\n",
      "Get:8 http://archive.ubuntu.com/ubuntu focal/universe amd64 Packages [11.3 MB]\n",
      "Get:9 http://security.ubuntu.com/ubuntu focal-security/main amd64 Packages [3195 kB]\n",
      "Get:10 http://security.ubuntu.com/ubuntu focal-security/restricted amd64 Packages [3017 kB]\n",
      "Get:11 http://archive.ubuntu.com/ubuntu focal/main amd64 Packages [1275 kB]    \n",
      "Get:12 http://archive.ubuntu.com/ubuntu focal/restricted amd64 Packages [33.4 kB]\n",
      "Get:13 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 Packages [1429 kB]\n",
      "Get:14 http://archive.ubuntu.com/ubuntu focal-updates/restricted amd64 Packages [3166 kB]\n",
      "Get:15 http://archive.ubuntu.com/ubuntu focal-updates/multiverse amd64 Packages [32.0 kB]\n",
      "Get:16 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 Packages [3683 kB]\n",
      "Get:17 http://archive.ubuntu.com/ubuntu focal-backports/main amd64 Packages [55.2 kB]\n",
      "Get:18 http://archive.ubuntu.com/ubuntu focal-backports/universe amd64 Packages [28.6 kB]\n",
      "Fetched 29.2 MB in 6s (4665 kB/s)                                              \n",
      "Reading package lists... Done\n",
      "Reading package lists... Done\n",
      "Building dependency tree       \n",
      "Reading state information... Done\n",
      "The following additional packages will be installed:\n",
      "  libdrm-amdgpu1 libdrm-common libdrm-intel1 libdrm-nouveau2 libdrm-radeon1\n",
      "  libdrm2 libelf1 libgl1-mesa-dri libglapi-mesa libglvnd0 libglx-mesa0 libglx0\n",
      "  libllvm12 libpciaccess0 libsensors-config libsensors5 libvulkan1\n",
      "  libwayland-client0 libx11-xcb1 libxcb-dri2-0 libxcb-dri3-0 libxcb-glx0\n",
      "  libxcb-present0 libxcb-randr0 libxcb-shm0 libxcb-sync1 libxcb-xfixes0\n",
      "  libxfixes3 libxshmfence1 libxxf86vm1 mesa-vulkan-drivers\n",
      "Suggested packages:\n",
      "  pciutils lm-sensors\n",
      "The following NEW packages will be installed:\n",
      "  libdrm-amdgpu1 libdrm-common libdrm-intel1 libdrm-nouveau2 libdrm-radeon1\n",
      "  libdrm2 libelf1 libgl1 libgl1-mesa-dri libglapi-mesa libglvnd0 libglx-mesa0\n",
      "  libglx0 libllvm12 libpciaccess0 libsensors-config libsensors5 libvulkan1\n",
      "  libwayland-client0 libx11-xcb1 libxcb-dri2-0 libxcb-dri3-0 libxcb-glx0\n",
      "  libxcb-present0 libxcb-randr0 libxcb-shm0 libxcb-sync1 libxcb-xfixes0\n",
      "  libxfixes3 libxshmfence1 libxxf86vm1 mesa-vulkan-drivers\n",
      "0 upgraded, 32 newly installed, 0 to remove and 194 not upgraded.\n",
      "Need to get 36.4 MB of archives.\n",
      "After this operation, 525 MB of additional disk space will be used.\n",
      "Get:1 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libelf1 amd64 0.176-1.1ubuntu0.1 [44.2 kB]\n",
      "Get:2 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libdrm-common all 2.4.107-8ubuntu1~20.04.2 [5396 B]\n",
      "Get:3 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libdrm2 amd64 2.4.107-8ubuntu1~20.04.2 [34.1 kB]\n",
      "Get:4 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libdrm-amdgpu1 amd64 2.4.107-8ubuntu1~20.04.2 [18.6 kB]\n",
      "Get:5 http://archive.ubuntu.com/ubuntu focal/main amd64 libpciaccess0 amd64 0.16-0ubuntu1 [17.9 kB]\n",
      "Get:6 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libdrm-intel1 amd64 2.4.107-8ubuntu1~20.04.2 [60.3 kB]\n",
      "Get:7 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libdrm-nouveau2 amd64 2.4.107-8ubuntu1~20.04.2 [16.6 kB]\n",
      "Get:8 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libdrm-radeon1 amd64 2.4.107-8ubuntu1~20.04.2 [19.7 kB]\n",
      "Get:9 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libglapi-mesa amd64 21.2.6-0ubuntu0.1~20.04.2 [27.4 kB]\n",
      "Get:10 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libllvm12 amd64 1:12.0.0-3ubuntu1~20.04.5 [18.8 MB]\n",
      "Get:11 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libsensors-config all 1:3.6.0-2ubuntu1.1 [6052 B]\n",
      "Get:12 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libsensors5 amd64 1:3.6.0-2ubuntu1.1 [27.2 kB]\n",
      "Get:13 http://archive.ubuntu.com/ubuntu focal/main amd64 libvulkan1 amd64 1.2.131.2-1 [93.3 kB]\n",
      "Get:14 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libgl1-mesa-dri amd64 21.2.6-0ubuntu0.1~20.04.2 [11.0 MB]\n",
      "Get:15 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libx11-xcb1 amd64 2:1.6.9-2ubuntu1.6 [9448 B]\n",
      "Get:16 http://archive.ubuntu.com/ubuntu focal/main amd64 libxcb-dri2-0 amd64 1.14-2 [6920 B]\n",
      "Get:17 http://archive.ubuntu.com/ubuntu focal/main amd64 libxcb-dri3-0 amd64 1.14-2 [6552 B]\n",
      "Get:18 http://archive.ubuntu.com/ubuntu focal/main amd64 libxcb-glx0 amd64 1.14-2 [22.1 kB]\n",
      "Get:19 http://archive.ubuntu.com/ubuntu focal/main amd64 libxcb-present0 amd64 1.14-2 [5560 B]\n",
      "Get:20 http://archive.ubuntu.com/ubuntu focal/main amd64 libxcb-shm0 amd64 1.14-2 [5584 B]\n",
      "Get:21 http://archive.ubuntu.com/ubuntu focal/main amd64 libxcb-sync1 amd64 1.14-2 [8884 B]\n",
      "Get:22 http://archive.ubuntu.com/ubuntu focal/main amd64 libxcb-xfixes0 amd64 1.14-2 [9296 B]\n",
      "Get:23 http://archive.ubuntu.com/ubuntu focal/main amd64 libxfixes3 amd64 1:5.0.3-2 [10.9 kB]\n",
      "Get:24 http://archive.ubuntu.com/ubuntu focal/main amd64 libxshmfence1 amd64 1.3-1 [5028 B]\n",
      "Get:25 http://archive.ubuntu.com/ubuntu focal/main amd64 libxxf86vm1 amd64 1:1.1.4-1build1 [10.2 kB]\n",
      "Get:26 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libglx-mesa0 amd64 21.2.6-0ubuntu0.1~20.04.2 [137 kB]\n",
      "Get:27 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libwayland-client0 amd64 1.18.0-1ubuntu0.1 [23.9 kB]\n",
      "Get:28 http://archive.ubuntu.com/ubuntu focal/main amd64 libxcb-randr0 amd64 1.14-2 [16.3 kB]\n",
      "Get:29 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 mesa-vulkan-drivers amd64 21.2.6-0ubuntu0.1~20.04.2 [5788 kB]\n",
      "Get:30 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libglvnd0 amd64 1.3.2-1~ubuntu0.20.04.2 [48.1 kB]\n",
      "Get:31 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libglx0 amd64 1.3.2-1~ubuntu0.20.04.2 [32.5 kB]\n",
      "Get:32 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libgl1 amd64 1.3.2-1~ubuntu0.20.04.2 [85.8 kB]\n",
      "Fetched 36.4 MB in 10s (3741 kB/s)                                             \n",
      "debconf: unable to initialize frontend: Dialog\n",
      "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 32.)\n",
      "debconf: falling back to frontend: Readline\n",
      "Extracting templates from packages: 100%\n",
      "Selecting previously unselected package libelf1:amd64.\n",
      "(Reading database ... 40531 files and directories currently installed.)\n",
      "Preparing to unpack .../00-libelf1_0.176-1.1ubuntu0.1_amd64.deb ...\n",
      "Unpacking libelf1:amd64 (0.176-1.1ubuntu0.1) ...\n",
      "Selecting previously unselected package libdrm-common.\n",
      "Preparing to unpack .../01-libdrm-common_2.4.107-8ubuntu1~20.04.2_all.deb ...\n",
      "Unpacking libdrm-common (2.4.107-8ubuntu1~20.04.2) ...\n",
      "Selecting previously unselected package libdrm2:amd64.\n",
      "Preparing to unpack .../02-libdrm2_2.4.107-8ubuntu1~20.04.2_amd64.deb ...\n",
      "Unpacking libdrm2:amd64 (2.4.107-8ubuntu1~20.04.2) ...\n",
      "Selecting previously unselected package libdrm-amdgpu1:amd64.\n",
      "Preparing to unpack .../03-libdrm-amdgpu1_2.4.107-8ubuntu1~20.04.2_amd64.deb ...\n",
      "Unpacking libdrm-amdgpu1:amd64 (2.4.107-8ubuntu1~20.04.2) ...\n",
      "Selecting previously unselected package libpciaccess0:amd64.\n",
      "Preparing to unpack .../04-libpciaccess0_0.16-0ubuntu1_amd64.deb ...\n",
      "Unpacking libpciaccess0:amd64 (0.16-0ubuntu1) ...\n",
      "Selecting previously unselected package libdrm-intel1:amd64.\n",
      "Preparing to unpack .../05-libdrm-intel1_2.4.107-8ubuntu1~20.04.2_amd64.deb ...\n",
      "Unpacking libdrm-intel1:amd64 (2.4.107-8ubuntu1~20.04.2) ...\n",
      "Selecting previously unselected package libdrm-nouveau2:amd64.\n",
      "Preparing to unpack .../06-libdrm-nouveau2_2.4.107-8ubuntu1~20.04.2_amd64.deb ...\n",
      "Unpacking libdrm-nouveau2:amd64 (2.4.107-8ubuntu1~20.04.2) ...\n",
      "Selecting previously unselected package libdrm-radeon1:amd64.\n",
      "Preparing to unpack .../07-libdrm-radeon1_2.4.107-8ubuntu1~20.04.2_amd64.deb ...\n",
      "Unpacking libdrm-radeon1:amd64 (2.4.107-8ubuntu1~20.04.2) ...\n",
      "Selecting previously unselected package libglapi-mesa:amd64.\n",
      "Preparing to unpack .../08-libglapi-mesa_21.2.6-0ubuntu0.1~20.04.2_amd64.deb ...\n",
      "Unpacking libglapi-mesa:amd64 (21.2.6-0ubuntu0.1~20.04.2) ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Selecting previously unselected package libllvm12:amd64.\n",
      "Preparing to unpack .../09-libllvm12_1%3a12.0.0-3ubuntu1~20.04.5_amd64.deb ...\n",
      "Unpacking libllvm12:amd64 (1:12.0.0-3ubuntu1~20.04.5) ...\n",
      "Selecting previously unselected package libsensors-config.\n",
      "Preparing to unpack .../10-libsensors-config_1%3a3.6.0-2ubuntu1.1_all.deb ...\n",
      "Unpacking libsensors-config (1:3.6.0-2ubuntu1.1) ...\n",
      "Selecting previously unselected package libsensors5:amd64.\n",
      "Preparing to unpack .../11-libsensors5_1%3a3.6.0-2ubuntu1.1_amd64.deb ...\n",
      "Unpacking libsensors5:amd64 (1:3.6.0-2ubuntu1.1) ...\n",
      "Selecting previously unselected package libvulkan1:amd64.\n",
      "Preparing to unpack .../12-libvulkan1_1.2.131.2-1_amd64.deb ...\n",
      "Unpacking libvulkan1:amd64 (1.2.131.2-1) ...\n",
      "Selecting previously unselected package libgl1-mesa-dri:amd64.\n",
      "Preparing to unpack .../13-libgl1-mesa-dri_21.2.6-0ubuntu0.1~20.04.2_amd64.deb ...\n",
      "Unpacking libgl1-mesa-dri:amd64 (21.2.6-0ubuntu0.1~20.04.2) ...\n",
      "Selecting previously unselected package libx11-xcb1:amd64.\n",
      "Preparing to unpack .../14-libx11-xcb1_2%3a1.6.9-2ubuntu1.6_amd64.deb ...\n",
      "Unpacking libx11-xcb1:amd64 (2:1.6.9-2ubuntu1.6) ...\n",
      "Selecting previously unselected package libxcb-dri2-0:amd64.\n",
      "Preparing to unpack .../15-libxcb-dri2-0_1.14-2_amd64.deb ...\n",
      "Unpacking libxcb-dri2-0:amd64 (1.14-2) ...\n",
      "Selecting previously unselected package libxcb-dri3-0:amd64.\n",
      "Preparing to unpack .../16-libxcb-dri3-0_1.14-2_amd64.deb ...\n",
      "Unpacking libxcb-dri3-0:amd64 (1.14-2) ...\n",
      "Selecting previously unselected package libxcb-glx0:amd64.\n",
      "Preparing to unpack .../17-libxcb-glx0_1.14-2_amd64.deb ...\n",
      "Unpacking libxcb-glx0:amd64 (1.14-2) ...\n",
      "Selecting previously unselected package libxcb-present0:amd64.\n",
      "Preparing to unpack .../18-libxcb-present0_1.14-2_amd64.deb ...\n",
      "Unpacking libxcb-present0:amd64 (1.14-2) ...\n",
      "Selecting previously unselected package libxcb-shm0:amd64.\n",
      "Preparing to unpack .../19-libxcb-shm0_1.14-2_amd64.deb ...\n",
      "Unpacking libxcb-shm0:amd64 (1.14-2) ...\n",
      "Selecting previously unselected package libxcb-sync1:amd64.\n",
      "Preparing to unpack .../20-libxcb-sync1_1.14-2_amd64.deb ...\n",
      "Unpacking libxcb-sync1:amd64 (1.14-2) ...\n",
      "Selecting previously unselected package libxcb-xfixes0:amd64.\n",
      "Preparing to unpack .../21-libxcb-xfixes0_1.14-2_amd64.deb ...\n",
      "Unpacking libxcb-xfixes0:amd64 (1.14-2) ...\n",
      "Selecting previously unselected package libxfixes3:amd64.\n",
      "Preparing to unpack .../22-libxfixes3_1%3a5.0.3-2_amd64.deb ...\n",
      "Unpacking libxfixes3:amd64 (1:5.0.3-2) ...\n",
      "Selecting previously unselected package libxshmfence1:amd64.\n",
      "Preparing to unpack .../23-libxshmfence1_1.3-1_amd64.deb ...\n",
      "Unpacking libxshmfence1:amd64 (1.3-1) ...\n",
      "Selecting previously unselected package libxxf86vm1:amd64.\n",
      "Preparing to unpack .../24-libxxf86vm1_1%3a1.1.4-1build1_amd64.deb ...\n",
      "Unpacking libxxf86vm1:amd64 (1:1.1.4-1build1) ...\n",
      "Selecting previously unselected package libglx-mesa0:amd64.\n",
      "Preparing to unpack .../25-libglx-mesa0_21.2.6-0ubuntu0.1~20.04.2_amd64.deb ...\n",
      "Unpacking libglx-mesa0:amd64 (21.2.6-0ubuntu0.1~20.04.2) ...\n",
      "Selecting previously unselected package libwayland-client0:amd64.\n",
      "Preparing to unpack .../26-libwayland-client0_1.18.0-1ubuntu0.1_amd64.deb ...\n",
      "Unpacking libwayland-client0:amd64 (1.18.0-1ubuntu0.1) ...\n",
      "Selecting previously unselected package libxcb-randr0:amd64.\n",
      "Preparing to unpack .../27-libxcb-randr0_1.14-2_amd64.deb ...\n",
      "Unpacking libxcb-randr0:amd64 (1.14-2) ...\n",
      "Selecting previously unselected package mesa-vulkan-drivers:amd64.\n",
      "Preparing to unpack .../28-mesa-vulkan-drivers_21.2.6-0ubuntu0.1~20.04.2_amd64.deb ...\n",
      "Unpacking mesa-vulkan-drivers:amd64 (21.2.6-0ubuntu0.1~20.04.2) ...\n",
      "Selecting previously unselected package libglvnd0:amd64.\n",
      "Preparing to unpack .../29-libglvnd0_1.3.2-1~ubuntu0.20.04.2_amd64.deb ...\n",
      "Unpacking libglvnd0:amd64 (1.3.2-1~ubuntu0.20.04.2) ...\n",
      "Selecting previously unselected package libglx0:amd64.\n",
      "Preparing to unpack .../30-libglx0_1.3.2-1~ubuntu0.20.04.2_amd64.deb ...\n",
      "Unpacking libglx0:amd64 (1.3.2-1~ubuntu0.20.04.2) ...\n",
      "Selecting previously unselected package libgl1:amd64.\n",
      "Preparing to unpack .../31-libgl1_1.3.2-1~ubuntu0.20.04.2_amd64.deb ...\n",
      "Unpacking libgl1:amd64 (1.3.2-1~ubuntu0.20.04.2) ...\n",
      "Setting up libxcb-dri3-0:amd64 (1.14-2) ...\n",
      "Setting up libx11-xcb1:amd64 (2:1.6.9-2ubuntu1.6) ...\n",
      "Setting up libpciaccess0:amd64 (0.16-0ubuntu1) ...\n",
      "Setting up libxcb-xfixes0:amd64 (1.14-2) ...\n",
      "Setting up libglvnd0:amd64 (1.3.2-1~ubuntu0.20.04.2) ...\n",
      "Setting up libxcb-glx0:amd64 (1.14-2) ...\n",
      "Setting up libsensors-config (1:3.6.0-2ubuntu1.1) ...\n",
      "Setting up libxcb-shm0:amd64 (1.14-2) ...\n",
      "Setting up libxxf86vm1:amd64 (1:1.1.4-1build1) ...\n",
      "Setting up libxcb-present0:amd64 (1.14-2) ...\n",
      "Setting up libxfixes3:amd64 (1:5.0.3-2) ...\n",
      "Setting up libxcb-sync1:amd64 (1.14-2) ...\n",
      "Setting up libllvm12:amd64 (1:12.0.0-3ubuntu1~20.04.5) ...\n",
      "Setting up libsensors5:amd64 (1:3.6.0-2ubuntu1.1) ...\n",
      "Setting up libglapi-mesa:amd64 (21.2.6-0ubuntu0.1~20.04.2) ...\n",
      "Setting up libvulkan1:amd64 (1.2.131.2-1) ...\n",
      "Setting up libxcb-dri2-0:amd64 (1.14-2) ...\n",
      "Setting up libxshmfence1:amd64 (1.3-1) ...\n",
      "Setting up libxcb-randr0:amd64 (1.14-2) ...\n",
      "Setting up libdrm-common (2.4.107-8ubuntu1~20.04.2) ...\n",
      "Setting up libelf1:amd64 (0.176-1.1ubuntu0.1) ...\n",
      "Setting up libwayland-client0:amd64 (1.18.0-1ubuntu0.1) ...\n",
      "Setting up libdrm2:amd64 (2.4.107-8ubuntu1~20.04.2) ...\n",
      "Setting up libdrm-amdgpu1:amd64 (2.4.107-8ubuntu1~20.04.2) ...\n",
      "Setting up mesa-vulkan-drivers:amd64 (21.2.6-0ubuntu0.1~20.04.2) ...\n",
      "Setting up libdrm-nouveau2:amd64 (2.4.107-8ubuntu1~20.04.2) ...\n",
      "Setting up libdrm-radeon1:amd64 (2.4.107-8ubuntu1~20.04.2) ...\n",
      "Setting up libdrm-intel1:amd64 (2.4.107-8ubuntu1~20.04.2) ...\n",
      "Setting up libgl1-mesa-dri:amd64 (21.2.6-0ubuntu0.1~20.04.2) ...\n",
      "Setting up libglx-mesa0:amd64 (21.2.6-0ubuntu0.1~20.04.2) ...\n",
      "Setting up libglx0:amd64 (1.3.2-1~ubuntu0.20.04.2) ...\n",
      "Setting up libgl1:amd64 (1.3.2-1~ubuntu0.20.04.2) ...\n",
      "Processing triggers for libc-bin (2.31-0ubuntu9.2) ...\n"
     ]
    }
   ],
   "source": [
    "!sudo apt-get update\n",
    "!sudo apt-get install libgl1 -y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "1NZxhXTMWvek",
    "outputId": "69470765-d772-45f9-d1a3-ee44ffeb7663"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Looking in links: https://download.pytorch.org/whl/cu113/torch_stable.html\n",
      "Requirement already satisfied: torch==1.10.1+cu113 in /home/u7701783/.local/lib/python3.8/site-packages (1.10.1+cu113)\n",
      "Requirement already satisfied: torchvision==0.11.2+cu113 in /home/u7701783/.local/lib/python3.8/site-packages (0.11.2+cu113)\n",
      "Requirement already satisfied: torchaudio==0.10.1+cu113 in /home/u7701783/.local/lib/python3.8/site-packages (0.10.1+cu113)\n",
      "Requirement already satisfied: typing-extensions in /home/u7701783/.local/lib/python3.8/site-packages (from torch==1.10.1+cu113) (4.8.0)\n",
      "Requirement already satisfied: numpy in /home/u7701783/.local/lib/python3.8/site-packages (from torchvision==0.11.2+cu113) (1.23.1)\n",
      "Requirement already satisfied: pillow!=8.3.0,>=5.3.0 in /home/u7701783/.local/lib/python3.8/site-packages (from torchvision==0.11.2+cu113) (10.1.0)\n",
      "\u001b[33mWARNING: You are using pip version 21.2.4; however, version 23.3.1 is available.\n",
      "You should consider upgrading via the '/opt/conda/bin/python3 -m pip install --upgrade pip' command.\u001b[0m\n",
      "nvcc: NVIDIA (R) Cuda compiler driver\n",
      "Copyright (c) 2005-2021 NVIDIA Corporation\n",
      "Built on Wed_Jul_14_19:41:19_PDT_2021\n",
      "Cuda compilation tools, release 11.4, V11.4.100\n",
      "Build cuda_11.4.r11.4/compiler.30188945_0\n",
      "Tue Nov 14 18:14:19 2023       \n",
      "+-----------------------------------------------------------------------------+\n",
      "| NVIDIA-SMI 470.161.03   Driver Version: 470.161.03   CUDA Version: 11.4     |\n",
      "|-------------------------------+----------------------+----------------------+\n",
      "| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |\n",
      "| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |\n",
      "|                               |                      |               MIG M. |\n",
      "|===============================+======================+======================|\n",
      "|   0  Tesla V100-SXM2...  On   | 00000000:B1:00.0 Off |                    0 |\n",
      "| N/A   26C    P0    42W / 300W |      0MiB / 32510MiB |      0%      Default |\n",
      "|                               |                      |                  N/A |\n",
      "+-------------------------------+----------------------+----------------------+\n",
      "                                                                               \n",
      "+-----------------------------------------------------------------------------+\n",
      "| Processes:                                                                  |\n",
      "|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |\n",
      "|        ID   ID                                                   Usage      |\n",
      "|=============================================================================|\n",
      "|  No running processes found                                                 |\n",
      "+-----------------------------------------------------------------------------+\n"
     ]
    }
   ],
   "source": [
    "!pip3 install torch==1.10.1+cu113 torchvision==0.11.2+cu113 torchaudio==0.10.1+cu113 -f https://download.pytorch.org/whl/cu113/torch_stable.html\n",
    "!nvcc -V\n",
    "!nvidia-smi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "9cWpOA-iLBdJ",
    "outputId": "7e77a653-4371-45f5-8848-0a7bc1da8690"
   },
   "outputs": [],
   "source": [
    "#!git clone git@github.com:Suppersine/YOLOv7_obb_KFIOU.git\n",
    "#!git clone https://github.com/SSTato/YOLOv7_obb.git"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "XrMDF3vubJlt",
    "outputId": "4b7d6529-86f5-4798-e362-d7826eeba68d"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/u7701783/yolov7-obb_shchen\n",
      "Requirement already satisfied: matplotlib>=3.2.2 in /home/u7701783/.local/lib/python3.8/site-packages (from -r requirements.txt (line 4)) (3.4.3)\n",
      "Requirement already satisfied: numpy>=1.18.5 in /home/u7701783/.local/lib/python3.8/site-packages (from -r requirements.txt (line 5)) (1.23.1)\n",
      "Requirement already satisfied: opencv-python>=4.5.4 in /home/u7701783/.local/lib/python3.8/site-packages (from -r requirements.txt (line 6)) (4.7.0.72)\n",
      "Requirement already satisfied: Pillow>=7.1.2 in /home/u7701783/.local/lib/python3.8/site-packages (from -r requirements.txt (line 7)) (10.1.0)\n",
      "Requirement already satisfied: PyYAML>=5.3.1 in /home/u7701783/.local/lib/python3.8/site-packages (from -r requirements.txt (line 8)) (5.4.1)\n",
      "Requirement already satisfied: requests>=2.23.0 in /opt/conda/lib/python3.8/site-packages (from -r requirements.txt (line 9)) (2.26.0)\n",
      "Requirement already satisfied: scipy>=1.4.1 in /home/u7701783/.local/lib/python3.8/site-packages (from -r requirements.txt (line 10)) (1.9.1)\n",
      "Requirement already satisfied: torch>=1.7.0 in /home/u7701783/.local/lib/python3.8/site-packages (from -r requirements.txt (line 11)) (1.10.1+cu113)\n",
      "Requirement already satisfied: torchvision>=0.8.1 in /home/u7701783/.local/lib/python3.8/site-packages (from -r requirements.txt (line 12)) (0.11.2+cu113)\n",
      "Requirement already satisfied: tqdm>=4.41.0 in /home/u7701783/.local/lib/python3.8/site-packages (from -r requirements.txt (line 13)) (4.66.1)\n",
      "Requirement already satisfied: tensorboard>=2.4.1 in /opt/conda/lib/python3.8/site-packages (from -r requirements.txt (line 16)) (2.6.0)\n",
      "Requirement already satisfied: pandas>=1.1.4 in /home/u7701783/.local/lib/python3.8/site-packages (from -r requirements.txt (line 20)) (1.3.5)\n",
      "Requirement already satisfied: seaborn>=0.11.0 in /home/u7701783/.local/lib/python3.8/site-packages (from -r requirements.txt (line 21)) (0.11.2)\n",
      "Requirement already satisfied: thop in /home/u7701783/.local/lib/python3.8/site-packages (from -r requirements.txt (line 37)) (0.1.1.post2209072238)\n",
      "Requirement already satisfied: pyparsing>=2.2.1 in /home/u7701783/.local/lib/python3.8/site-packages (from matplotlib>=3.2.2->-r requirements.txt (line 4)) (2.4.7)\n",
      "Requirement already satisfied: cycler>=0.10 in /home/u7701783/.local/lib/python3.8/site-packages (from matplotlib>=3.2.2->-r requirements.txt (line 4)) (0.10.0)\n",
      "Requirement already satisfied: python-dateutil>=2.7 in /opt/conda/lib/python3.8/site-packages (from matplotlib>=3.2.2->-r requirements.txt (line 4)) (2.8.2)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in /home/u7701783/.local/lib/python3.8/site-packages (from matplotlib>=3.2.2->-r requirements.txt (line 4)) (1.3.1)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /home/u7701783/.local/lib/python3.8/site-packages (from requests>=2.23.0->-r requirements.txt (line 9)) (2021.5.30)\n",
      "Requirement already satisfied: charset-normalizer~=2.0.0 in /opt/conda/lib/python3.8/site-packages (from requests>=2.23.0->-r requirements.txt (line 9)) (2.0.0)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /home/u7701783/.local/lib/python3.8/site-packages (from requests>=2.23.0->-r requirements.txt (line 9)) (1.26.6)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /home/u7701783/.local/lib/python3.8/site-packages (from requests>=2.23.0->-r requirements.txt (line 9)) (2.10)\n",
      "Requirement already satisfied: typing-extensions in /home/u7701783/.local/lib/python3.8/site-packages (from torch>=1.7.0->-r requirements.txt (line 11)) (4.8.0)\n",
      "Requirement already satisfied: setuptools>=41.0.0 in /home/u7701783/.local/lib/python3.8/site-packages (from tensorboard>=2.4.1->-r requirements.txt (line 16)) (59.5.0)\n",
      "Requirement already satisfied: wheel>=0.26 in /opt/conda/lib/python3.8/site-packages (from tensorboard>=2.4.1->-r requirements.txt (line 16)) (0.36.2)\n",
      "Requirement already satisfied: grpcio>=1.24.3 in /opt/conda/lib/python3.8/site-packages (from tensorboard>=2.4.1->-r requirements.txt (line 16)) (1.39.0)\n",
      "Requirement already satisfied: werkzeug>=0.11.15 in /opt/conda/lib/python3.8/site-packages (from tensorboard>=2.4.1->-r requirements.txt (line 16)) (2.0.1)\n",
      "Requirement already satisfied: absl-py>=0.4 in /opt/conda/lib/python3.8/site-packages (from tensorboard>=2.4.1->-r requirements.txt (line 16)) (0.13.0)\n",
      "Requirement already satisfied: markdown>=2.6.8 in /opt/conda/lib/python3.8/site-packages (from tensorboard>=2.4.1->-r requirements.txt (line 16)) (3.3.4)\n",
      "Requirement already satisfied: tensorboard-plugin-wit>=1.6.0 in /opt/conda/lib/python3.8/site-packages (from tensorboard>=2.4.1->-r requirements.txt (line 16)) (1.8.0)\n",
      "Requirement already satisfied: google-auth-oauthlib<0.5,>=0.4.1 in /opt/conda/lib/python3.8/site-packages (from tensorboard>=2.4.1->-r requirements.txt (line 16)) (0.4.5)\n",
      "Requirement already satisfied: google-auth<2,>=1.6.3 in /opt/conda/lib/python3.8/site-packages (from tensorboard>=2.4.1->-r requirements.txt (line 16)) (1.34.0)\n",
      "Requirement already satisfied: tensorboard-data-server<0.7.0,>=0.6.0 in /opt/conda/lib/python3.8/site-packages (from tensorboard>=2.4.1->-r requirements.txt (line 16)) (0.6.1)\n",
      "Requirement already satisfied: protobuf>=3.6.0 in /opt/conda/lib/python3.8/site-packages (from tensorboard>=2.4.1->-r requirements.txt (line 16)) (3.17.3)\n",
      "Requirement already satisfied: pytz>=2017.3 in /opt/conda/lib/python3.8/site-packages (from pandas>=1.1.4->-r requirements.txt (line 20)) (2021.1)\n",
      "Requirement already satisfied: six in /opt/conda/lib/python3.8/site-packages (from absl-py>=0.4->tensorboard>=2.4.1->-r requirements.txt (line 16)) (1.16.0)\n",
      "Requirement already satisfied: rsa<5,>=3.1.4 in /opt/conda/lib/python3.8/site-packages (from google-auth<2,>=1.6.3->tensorboard>=2.4.1->-r requirements.txt (line 16)) (4.7.2)\n",
      "Requirement already satisfied: cachetools<5.0,>=2.0.0 in /opt/conda/lib/python3.8/site-packages (from google-auth<2,>=1.6.3->tensorboard>=2.4.1->-r requirements.txt (line 16)) (4.2.2)\n",
      "Requirement already satisfied: pyasn1-modules>=0.2.1 in /opt/conda/lib/python3.8/site-packages (from google-auth<2,>=1.6.3->tensorboard>=2.4.1->-r requirements.txt (line 16)) (0.2.8)\n",
      "Requirement already satisfied: requests-oauthlib>=0.7.0 in /opt/conda/lib/python3.8/site-packages (from google-auth-oauthlib<0.5,>=0.4.1->tensorboard>=2.4.1->-r requirements.txt (line 16)) (1.3.0)\n",
      "Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in /opt/conda/lib/python3.8/site-packages (from pyasn1-modules>=0.2.1->google-auth<2,>=1.6.3->tensorboard>=2.4.1->-r requirements.txt (line 16)) (0.4.8)\n",
      "Requirement already satisfied: oauthlib>=3.0.0 in /opt/conda/lib/python3.8/site-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<0.5,>=0.4.1->tensorboard>=2.4.1->-r requirements.txt (line 16)) (3.1.1)\n",
      "\u001b[33mWARNING: You are using pip version 21.2.4; however, version 23.3.1 is available.\n",
      "You should consider upgrading via the '/opt/conda/bin/python3 -m pip install --upgrade pip' command.\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#%cd /content/yolov7_obb_expm/\n",
    "%cd ~/yolov7-obb_shchen/\n",
    "!pip install -r requirements.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "11G939jgNaVK",
    "outputId": "1516081a-a8d1-411f-e509-e92213e9b121"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/u7701783/yolov7-obb_shchen/utils/nms_rotated\n",
      "running develop\n",
      "/home/u7701783/.local/lib/python3.8/site-packages/setuptools/command/easy_install.py:156: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.\n",
      "  warnings.warn(\n",
      "/home/u7701783/.local/lib/python3.8/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.\n",
      "  warnings.warn(\n",
      "error: can't create or remove files in install directory\n",
      "\n",
      "The following error occurred while trying to add or remove files in the\n",
      "installation directory:\n",
      "\n",
      "    [Errno 13] Permission denied: '/opt/conda/lib/python3.8/site-packages/test-easy-install-797.write-test'\n",
      "\n",
      "The installation directory you specified (via --install-dir, --prefix, or\n",
      "the distutils default setting) was:\n",
      "\n",
      "    /opt/conda/lib/python3.8/site-packages/\n",
      "\n",
      "Perhaps your account does not have write access to this directory?  If the\n",
      "installation directory is a system-owned directory, you may need to sign in\n",
      "as the administrator or \"root\" account.  If you do not have administrative\n",
      "access to this machine, you may wish to choose a different installation\n",
      "directory, preferably one that is listed in your PYTHONPATH environment\n",
      "variable.\n",
      "\n",
      "For information on other options, you may wish to consult the\n",
      "documentation at:\n",
      "\n",
      "  https://setuptools.pypa.io/en/latest/deprecated/easy_install.html\n",
      "\n",
      "Please make the appropriate changes for your system and try again.\n",
      "\n",
      "Using pip 21.2.4 from /opt/conda/lib/python3.8/site-packages/pip (python 3.8)\n",
      "Obtaining file:///home/u7701783/yolov7-obb_shchen/utils/nms_rotated\n",
      "    Running command python setup.py egg_info\n",
      "    running egg_info\n",
      "    creating /tmp/pip-pip-egg-info-7xda2xbj/nms_rotated.egg-info\n",
      "    writing /tmp/pip-pip-egg-info-7xda2xbj/nms_rotated.egg-info/PKG-INFO\n",
      "    writing dependency_links to /tmp/pip-pip-egg-info-7xda2xbj/nms_rotated.egg-info/dependency_links.txt\n",
      "    writing top-level names to /tmp/pip-pip-egg-info-7xda2xbj/nms_rotated.egg-info/top_level.txt\n",
      "    writing manifest file '/tmp/pip-pip-egg-info-7xda2xbj/nms_rotated.egg-info/SOURCES.txt'\n",
      "    reading manifest file '/tmp/pip-pip-egg-info-7xda2xbj/nms_rotated.egg-info/SOURCES.txt'\n",
      "    writing manifest file '/tmp/pip-pip-egg-info-7xda2xbj/nms_rotated.egg-info/SOURCES.txt'\n",
      "Installing collected packages: nms-rotated\n",
      "  Attempting uninstall: nms-rotated\n",
      "    Found existing installation: nms-rotated 0.0.0\n",
      "    Uninstalling nms-rotated-0.0.0:\n",
      "      Removing file or directory /home/u7701783/.local/lib/python3.8/site-packages/nms-rotated.egg-link\n",
      "      Removing pth entries from /home/u7701783/.local/lib/python3.8/site-packages/easy-install.pth:\n",
      "      Removing entry: /home/u7701783/yolov5_obb_new/utils/nms_rotated\n",
      "      Successfully uninstalled nms-rotated-0.0.0\n",
      "  Running setup.py develop for nms-rotated\n",
      "    Running command /opt/conda/bin/python3 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/home/u7701783/yolov7-obb_shchen/utils/nms_rotated/setup.py'\"'\"'; __file__='\"'\"'/home/u7701783/yolov7-obb_shchen/utils/nms_rotated/setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' develop --no-deps --user --prefix=\n",
      "    running develop\n",
      "    running egg_info\n",
      "    writing nms_rotated.egg-info/PKG-INFO\n",
      "    writing dependency_links to nms_rotated.egg-info/dependency_links.txt\n",
      "    writing top-level names to nms_rotated.egg-info/top_level.txt\n",
      "    reading manifest file 'nms_rotated.egg-info/SOURCES.txt'\n",
      "    writing manifest file 'nms_rotated.egg-info/SOURCES.txt'\n",
      "    running build_ext\n",
      "    building '.nms_rotated_ext' extension\n",
      "    Emitting ninja build file /home/u7701783/yolov7-obb_shchen/utils/nms_rotated/build/temp.linux-x86_64-3.8/build.ninja...\n",
      "    Compiling objects...\n",
      "    Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)\n",
      "    /home/u7701783/.local/lib/python3.8/site-packages/setuptools/command/easy_install.py:156: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.\n",
      "      warnings.warn(\n",
      "    /home/u7701783/.local/lib/python3.8/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.\n",
      "      warnings.warn(\n",
      "    /home/u7701783/.local/lib/python3.8/site-packages/torch/utils/cpp_extension.py:782: UserWarning: The detected CUDA version (11.4) has a minor version mismatch with the version that was used to compile PyTorch (11.3). Most likely this shouldn't be a problem.\n",
      "      warnings.warn(CUDA_MISMATCH_WARN.format(cuda_str_version, torch.version.cuda))\n",
      "    ninja: no work to do.\n",
      "    g++ -pthread -shared -B /opt/conda/compiler_compat -L/opt/conda/lib -Wl,-rpath=/opt/conda/lib -Wl,--no-as-needed -Wl,--sysroot=/ /home/u7701783/yolov7-obb_shchen/utils/nms_rotated/build/temp.linux-x86_64-3.8/src/nms_rotated_cpu.o /home/u7701783/yolov7-obb_shchen/utils/nms_rotated/build/temp.linux-x86_64-3.8/src/nms_rotated_ext.o /home/u7701783/yolov7-obb_shchen/utils/nms_rotated/build/temp.linux-x86_64-3.8/src/nms_rotated_cuda.o /home/u7701783/yolov7-obb_shchen/utils/nms_rotated/build/temp.linux-x86_64-3.8/src/poly_nms_cuda.o -L/home/u7701783/.local/lib/python3.8/site-packages/torch/lib -L/usr/local/cuda/lib64 -lc10 -ltorch -ltorch_cpu -ltorch_python -lcudart -lc10_cuda -ltorch_cuda_cu -ltorch_cuda_cpp -o build/lib.linux-x86_64-3.8/nms_rotated_ext.cpython-38-x86_64-linux-gnu.so\n",
      "    copying build/lib.linux-x86_64-3.8/nms_rotated_ext.cpython-38-x86_64-linux-gnu.so ->\n",
      "    Creating /home/u7701783/.local/lib/python3.8/site-packages/nms-rotated.egg-link (link to .)\n",
      "    Adding nms-rotated 0.0.0 to easy-install.pth file\n",
      "\n",
      "    Installed /home/u7701783/yolov7-obb_shchen/utils/nms_rotated\n",
      "Successfully installed nms-rotated-0.0.0\n",
      "\u001b[33mWARNING: You are using pip version 21.2.4; however, version 23.3.1 is available.\n",
      "You should consider upgrading via the '/opt/conda/bin/python3 -m pip install --upgrade pip' command.\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "%cd utils/nms_rotated\n",
    "!python setup.py develop  #or \"\n",
    "!pip install -v -e ."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "HGf0HWV2Lgns"
   },
   "source": [
    "##Export\n",
    "\n",
    "Remember to export your annotated dataset following the steps shown [in the video tutorial](https://youtu.be/iRkCNo9-slY?t=136)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "cZJDvs5Bd59B"
   },
   "source": [
    "## Rename dataset name to roboflow\n",
    "\n",
    "After running the `!pip install` package, our python.train will expect the dataset name to be `roboflow` (with or without changing the code path below). Make sure to change your cloned dataset folder name to `roboflow` before running the cell below. \n",
    "\n",
    "In other words, make sure your dataset structure is:\n",
    "\n",
    "```\n",
    "parent\n",
    "├── yolov5\n",
    "└── datasets\n",
    "    └── roboflow\n",
    "        ├── test\n",
    "        ├── train\n",
    "        └── valid\n",
    "            ├── images\n",
    "                 |────1.jpg\n",
    "                 |────...\n",
    "                 └────10000.jpg\n",
    "            ├── labelTxt\n",
    "                 |────1.txt\n",
    "                 |────...\n",
    "                 └────10000.txt\n",
    "        └────README.roboflow.txt\n",
    "        └────data.yaml\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "XZd6BBwHOCME"
   },
   "source": [
    "Before training the model, always rename the dataset to \"roboflow\", and point its address to:\n",
    "\n",
    "```\n",
    "/content/datasets/roboflow/data.yaml\n",
    "```\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/u7701783/yolov7-obb_shchen\n",
      "LE90 angular definition mode selected\n",
      "\u001b[34m\u001b[1mval: \u001b[0mdata=/home/u7701783/datasets/roboflow/data.yaml, weights=['runs/train/yv7csl640i/weights/best.pt'], batch_size=32, imgsz=640, conf_thres=0.01, iou_thres=0.4, task=val, device=0, workers=8, single_cls=False, augment=False, verbose=False, save_txt=False, save_hybrid=False, save_conf=False, save_json=False, project=runs/val, name=yv7csl640val, exist_ok=True, half=False, dnn=False\n",
      "YOLOv5 🚀 bf4fc14 torch 1.10.1+cu113 CUDA:0 (Tesla V100-SXM2-32GB, 32510MiB)\n",
      "\n",
      "Fusing layers... \n",
      "RepConv.fuse_repvgg_block\n",
      "RepConv.fuse_repvgg_block\n",
      "RepConv.fuse_repvgg_block\n",
      "Model Summary: 306 layers, 37465381 parameters, 6194944 gradients, 106.3 GFLOPs\n",
      "\u001b[34m\u001b[1mval: \u001b[0mScanning '../datasets/roboflow/valid/labelTxt' images and labels...204 foun\u001b[0m\n",
      "\u001b[34m\u001b[1mval: \u001b[0mWARNING: ../datasets/roboflow/valid/images/amtpwqhramk=-video4.mp4-8.png: ignoring corrupt image/label: 'RCA' is not in list\n",
      "\u001b[34m\u001b[1mval: \u001b[0mNew cache created: ../datasets/roboflow/valid/labelTxt.cache\n",
      "/home/u7701783/.local/lib/python3.8/site-packages/torch/utils/data/dataloader.py:478: UserWarning: This DataLoader will create 8 worker processes in total. Our suggested max number of worker in current system is 4, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
      "  warnings.warn(_create_warning_msg(\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-6:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "Exception in thread Thread-7:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-8:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "Exception in thread Thread-9:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-10:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "Exception in thread Thread-11:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBB\n",
      "                 all        203        277      0.371      0.496      0.376       0.15\n",
      "yolov7csl_val folder name selected\n",
      "      LCA-brightness        203         29     0.0732       0.31     0.0678     0.0299\n",
      "      LCA-dilatation        203        230      0.427      0.426      0.349     0.0811\n",
      "      RCA-brightness        203          7      0.236      0.429        0.2      0.038\n",
      "      RCA-dilatation        203         11      0.749      0.818      0.888      0.452\n",
      "Speed: 0.2ms pre-process, 8.8ms inference, 1.7ms NMS per image at shape (32, 3, 640, 640)\n",
      "Results saved to \u001b[1mruns/val/yv7csl640val\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "%cd ~/yolov7-obb_shchen/\n",
    "!python val.py --data ~/datasets/roboflow/data.yaml --weights 'runs/train/yv7csl640i/weights/best.pt' --task 'test' --batch-size 32 --img 640 --device 0 --exist-ok --name 'yv7csl640val'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LE90 angular definition mode selected\n",
      "\u001b[34m\u001b[1mval: \u001b[0mdata=/home/u7701783/datasets/roboflow/data.yaml, weights=['runs/train/yv7kld640i/weights/best.pt'], batch_size=32, imgsz=640, conf_thres=0.01, iou_thres=0.4, task=val, device=0, workers=8, single_cls=False, augment=False, verbose=False, save_txt=False, save_hybrid=False, save_conf=False, save_json=False, project=runs/val, name=yv7kld640val, exist_ok=True, half=False, dnn=False\n",
      "YOLOv5 🚀 bf4fc14 torch 1.10.1+cu113 CUDA:0 (Tesla V100-SXM2-32GB, 32510MiB)\n",
      "\n",
      "Fusing layers... \n",
      "RepConv.fuse_repvgg_block\n",
      "RepConv.fuse_repvgg_block\n",
      "RepConv.fuse_repvgg_block\n",
      "Model Summary: 306 layers, 37465381 parameters, 6194944 gradients, 106.3 GFLOPs\n",
      "\u001b[34m\u001b[1mval: \u001b[0mScanning '../datasets/roboflow/valid/labelTxt.cache' images and labels... 2\u001b[0m\n",
      "\u001b[34m\u001b[1mval: \u001b[0mWARNING: ../datasets/roboflow/valid/images/amtpwqhramk=-video4.mp4-8.png: ignoring corrupt image/label: 'RCA' is not in list\n",
      "/home/u7701783/.local/lib/python3.8/site-packages/torch/utils/data/dataloader.py:478: UserWarning: This DataLoader will create 8 worker processes in total. Our suggested max number of worker in current system is 4, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
      "  warnings.warn(_create_warning_msg(\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-4:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "Exception in thread Thread-3:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self.run()\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-5:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "Exception in thread Thread-6:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-7:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "Exception in thread Thread-8:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBB\n",
      "                 all        203        277      0.181      0.657      0.259      0.106\n",
      "yolov7csl_val folder name selected\n",
      "      LCA-brightness        203         29      0.106      0.828      0.178     0.0604\n",
      "      LCA-dilatation        203        230      0.563      0.555       0.49      0.158\n",
      "      RCA-brightness        203          7     0.0124      0.429     0.0104    0.00436\n",
      "      RCA-dilatation        203         11     0.0428      0.818      0.359      0.202\n",
      "Speed: 0.2ms pre-process, 6.7ms inference, 2.0ms NMS per image at shape (32, 3, 640, 640)\n",
      "Results saved to \u001b[1mruns/val/yv7kld640val\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!python val.py --data ~/datasets/roboflow/data.yaml --weights 'runs/train/yv7kld640i/weights/best.pt' --task 'test' --batch-size 32 --img 640 --device 0 --exist-ok --name 'yv7kld640val'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LE90 angular definition mode selected\n",
      "\u001b[34m\u001b[1mval: \u001b[0mdata=/home/u7701783/datasets/roboflow/data.yaml, weights=['runs/train/yv7KFIOU640i/weights/best.pt'], batch_size=32, imgsz=640, conf_thres=0.01, iou_thres=0.4, task=val, device=0, workers=8, single_cls=False, augment=False, verbose=False, save_txt=False, save_hybrid=False, save_conf=False, save_json=False, project=runs/val, name=yv7kfiou640val, exist_ok=True, half=False, dnn=False\n",
      "YOLOv5 🚀 bf4fc14 torch 1.10.1+cu113 CUDA:0 (Tesla V100-SXM2-32GB, 32510MiB)\n",
      "\n",
      "Fusing layers... \n",
      "RepConv.fuse_repvgg_block\n",
      "RepConv.fuse_repvgg_block\n",
      "RepConv.fuse_repvgg_block\n",
      "Model Summary: 306 layers, 37465381 parameters, 6194944 gradients, 106.3 GFLOPs\n",
      "\u001b[34m\u001b[1mval: \u001b[0mScanning '../datasets/roboflow/valid/labelTxt.cache' images and labels... 2\u001b[0m\n",
      "\u001b[34m\u001b[1mval: \u001b[0mWARNING: ../datasets/roboflow/valid/images/amtpwqhramk=-video4.mp4-8.png: ignoring corrupt image/label: 'RCA' is not in list\n",
      "/home/u7701783/.local/lib/python3.8/site-packages/torch/utils/data/dataloader.py:478: UserWarning: This DataLoader will create 8 worker processes in total. Our suggested max number of worker in current system is 4, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
      "  warnings.warn(_create_warning_msg(\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-3:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "Exception in thread Thread-4:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-5:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "Exception in thread Thread-6:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-7:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "Exception in thread Thread-8:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBB\n",
      "                 all        203        277      0.299      0.508      0.316      0.133\n",
      "yolov7csl_val folder name selected\n",
      "      LCA-brightness        203         29      0.177      0.483      0.169     0.0639\n",
      "      LCA-dilatation        203        230      0.612      0.343      0.475      0.151\n",
      "      RCA-brightness        203          7     0.0192      0.571     0.0547     0.0215\n",
      "      RCA-dilatation        203         11      0.389      0.636      0.565      0.296\n",
      "Speed: 0.2ms pre-process, 6.7ms inference, 2.3ms NMS per image at shape (32, 3, 640, 640)\n",
      "Results saved to \u001b[1mruns/val/yv7kfiou640val\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!python val.py --data ~/datasets/roboflow/data.yaml --weights 'runs/train/yv7KFIOU640i/weights/best.pt' --task 'test' --batch-size 32 --img 640 --device 0 --exist-ok --name 'yv7kfiou640val'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LE90 angular definition mode selected\n",
      "\u001b[34m\u001b[1mval: \u001b[0mdata=/home/u7701783/datasets/roboflow/data.yaml, weights=['runs/train/yv7xcsl640p5/weights/best.pt'], batch_size=32, imgsz=640, conf_thres=0.01, iou_thres=0.4, task=val, device=0, workers=8, single_cls=False, augment=False, verbose=False, save_txt=False, save_hybrid=False, save_conf=False, save_json=False, project=runs/val, name=yv7xcsl640val, exist_ok=True, half=False, dnn=False\n",
      "YOLOv5 🚀 bf4fc14 torch 1.10.1+cu113 CUDA:0 (Tesla V100-SXM2-32GB, 32510MiB)\n",
      "\n",
      "Fusing layers... \n",
      "IDetect.fuse\n",
      "/home/u7701783/.local/lib/python3.8/site-packages/torch/functional.py:445: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at  ../aten/src/ATen/native/TensorShape.cpp:2157.)\n",
      "  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]\n",
      "Model Summary: 362 layers, 72015498 parameters, 0 gradients, 191.9 GFLOPs\n",
      "\u001b[34m\u001b[1mval: \u001b[0mScanning '../datasets/roboflow/valid/labelTxt.cache' images and labels... 2\u001b[0m\n",
      "\u001b[34m\u001b[1mval: \u001b[0mWARNING: ../datasets/roboflow/valid/images/amtpwqhramk=-video4.mp4-8.png: ignoring corrupt image/label: 'RCA' is not in list\n",
      "/home/u7701783/.local/lib/python3.8/site-packages/torch/utils/data/dataloader.py:478: UserWarning: This DataLoader will create 8 worker processes in total. Our suggested max number of worker in current system is 4, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
      "  warnings.warn(_create_warning_msg(\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-3:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "Exception in thread Thread-4:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-5:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "Exception in thread Thread-6:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-7:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "Exception in thread Thread-8:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBB\n",
      "                 all        203        277      0.396      0.697      0.417      0.205\n",
      "yolov7csl_val folder name selected\n",
      "      LCA-brightness        203         29      0.123      0.828      0.178      0.088\n",
      "      LCA-dilatation        203        230      0.517      0.674      0.499      0.158\n",
      "      RCA-brightness        203          7       0.29      0.286      0.196     0.0964\n",
      "      RCA-dilatation        203         11      0.656          1      0.792      0.477\n",
      "Speed: 0.1ms pre-process, 10.6ms inference, 1.9ms NMS per image at shape (32, 3, 640, 640)\n",
      "Results saved to \u001b[1mruns/val/yv7xcsl640val\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!python val.py --data ~/datasets/roboflow/data.yaml --weights 'runs/train/yv7xcsl640p5/weights/best.pt' --task 'test' --batch-size 32 --img 640 --device 0 --exist-ok --name 'yv7xcsl640val'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LE90 angular definition mode selected\n",
      "\u001b[34m\u001b[1mval: \u001b[0mdata=/home/u7701783/datasets/roboflow/data.yaml, weights=['runs/train/yv7xkld640p5/weights/best.pt'], batch_size=32, imgsz=640, conf_thres=0.01, iou_thres=0.4, task=val, device=0, workers=8, single_cls=False, augment=False, verbose=False, save_txt=False, save_hybrid=False, save_conf=False, save_json=False, project=runs/val, name=yv7xklf640val, exist_ok=True, half=False, dnn=False\n",
      "YOLOv5 🚀 bf4fc14 torch 1.10.1+cu113 CUDA:0 (Tesla V100-SXM2-32GB, 32510MiB)\n",
      "\n",
      "Fusing layers... \n",
      "IDetect.fuse\n",
      "/home/u7701783/.local/lib/python3.8/site-packages/torch/functional.py:445: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at  ../aten/src/ATen/native/TensorShape.cpp:2157.)\n",
      "  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]\n",
      "Model Summary: 362 layers, 72015498 parameters, 0 gradients, 191.9 GFLOPs\n",
      "\u001b[34m\u001b[1mval: \u001b[0mScanning '../datasets/roboflow/valid/labelTxt.cache' images and labels... 2\u001b[0m\n",
      "\u001b[34m\u001b[1mval: \u001b[0mWARNING: ../datasets/roboflow/valid/images/amtpwqhramk=-video4.mp4-8.png: ignoring corrupt image/label: 'RCA' is not in list\n",
      "/home/u7701783/.local/lib/python3.8/site-packages/torch/utils/data/dataloader.py:478: UserWarning: This DataLoader will create 8 worker processes in total. Our suggested max number of worker in current system is 4, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
      "  warnings.warn(_create_warning_msg(\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-3:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBB    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "Exception in thread Thread-4:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "Exception in thread Thread-5:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-6:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "Exception in thread Thread-7:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-8:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBB\n",
      "                 all        203        277      0.242      0.399      0.357      0.179\n",
      "yolov7csl_val folder name selected\n",
      "      LCA-brightness        203         29      0.323      0.621      0.345      0.199\n",
      "      LCA-dilatation        203        230          0          0      0.449      0.186\n",
      "      RCA-brightness        203          7     0.0509      0.429     0.0481     0.0153\n",
      "      RCA-dilatation        203         11      0.596      0.545      0.585      0.315\n",
      "Speed: 0.1ms pre-process, 10.6ms inference, 10.3ms NMS per image at shape (32, 3, 640, 640)\n",
      "Results saved to \u001b[1mruns/val/yv7xklf640val\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!python val.py --data ~/datasets/roboflow/data.yaml --weights 'runs/train/yv7xkld640p5/weights/best.pt' --task 'test' --batch-size 32 --img 640 --device 0 --exist-ok --name 'yv7xklf640val'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LE90 angular definition mode selected\n",
      "\u001b[34m\u001b[1mval: \u001b[0mdata=/home/u7701783/datasets/roboflow/data.yaml, weights=['runs/train/yv7xkfiou640p5/weights/best.pt'], batch_size=32, imgsz=640, conf_thres=0.01, iou_thres=0.4, task=val, device=0, workers=8, single_cls=False, augment=False, verbose=False, save_txt=False, save_hybrid=False, save_conf=False, save_json=False, project=runs/val, name=yv7xkfiou640cal, exist_ok=True, half=False, dnn=False\n",
      "YOLOv5 🚀 bf4fc14 torch 1.10.1+cu113 CUDA:0 (Tesla V100-SXM2-32GB, 32510MiB)\n",
      "\n",
      "Fusing layers... \n",
      "IDetect.fuse\n",
      "/home/u7701783/.local/lib/python3.8/site-packages/torch/functional.py:445: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at  ../aten/src/ATen/native/TensorShape.cpp:2157.)\n",
      "  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]\n",
      "Model Summary: 362 layers, 72015498 parameters, 0 gradients, 191.9 GFLOPs\n",
      "\u001b[34m\u001b[1mval: \u001b[0mScanning '../datasets/roboflow/valid/labelTxt.cache' images and labels... 2\u001b[0m\n",
      "\u001b[34m\u001b[1mval: \u001b[0mWARNING: ../datasets/roboflow/valid/images/amtpwqhramk=-video4.mp4-8.png: ignoring corrupt image/label: 'RCA' is not in list\n",
      "/home/u7701783/.local/lib/python3.8/site-packages/torch/utils/data/dataloader.py:478: UserWarning: This DataLoader will create 8 worker processes in total. Our suggested max number of worker in current system is 4, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.\n",
      "  warnings.warn(_create_warning_msg(\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-3:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-4:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "Exception in thread Thread-5:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-6:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "Exception in thread Thread-7:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBBException in thread Thread-8:\n",
      "Traceback (most recent call last):\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/opt/conda/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 267, in plot_images\n",
      "    annotator.text((x + 5, y + 5 + h), text=Path(paths[i]).name[:40], txt_color=(220, 220, 220))  # filenames\n",
      "  File \"/home/u7701783/yolov7-obb_shchen/utils/plots.py\", line 155, in text\n",
      "    w, h = self.font.getsize(text)  # text width, height\n",
      "AttributeError: 'FreeTypeFont' object has no attribute 'getsize'\n",
      "               Class     Images     Labels          P          R  HBBmAP@.5  HBB\n",
      "                 all        203        277      0.471      0.452      0.379      0.153\n",
      "yolov7csl_val folder name selected\n",
      "      LCA-brightness        203         29      0.222      0.448      0.191     0.0927\n",
      "      LCA-dilatation        203        230          1          0      0.408       0.12\n",
      "      RCA-brightness        203          7     0.0709      0.571      0.199     0.0668\n",
      "      RCA-dilatation        203         11       0.59      0.788      0.718      0.334\n",
      "Speed: 0.1ms pre-process, 10.6ms inference, 10.3ms NMS per image at shape (32, 3, 640, 640)\n",
      "Results saved to \u001b[1mruns/val/yv7xkfiou640cal\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!python val.py --data ~/datasets/roboflow/data.yaml --weights 'runs/train/yv7xkfiou640p5/weights/best.pt' --task 'test' --batch-size 32 --img 640 --device 0 --exist-ok --name 'yv7xkfiou640cal'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "id": "D782nBcaL5er"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[Errno 2] No such file or directory: '/content/yolov5_obb/runs/detect/exp/'\n",
      "/home/u7701783/yolov7-obb_shchen\n"
     ]
    }
   ],
   "source": [
    "%cd /content/yolov5_obb/runs/detect/exp/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "id": "K6RlZpzxb_kE"
   },
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/content/yolov5_obb/runs/detect/exp'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_2627/278909991.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrandom\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mrandom_file\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrandom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mchoice\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlistdir\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"/content/yolov5_obb/runs/detect/exp\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mIPython\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdisplay\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mImage\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mImage\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrandom_file\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/content/yolov5_obb/runs/detect/exp'"
     ]
    }
   ],
   "source": [
    "import os, random\n",
    "random_file = random.choice(os.listdir(\"/content/yolov5_obb/runs/detect/exp\"))\n",
    "\n",
    "from IPython.display import Image\n",
    "Image(random_file)"
   ]
  }
 ],
 "metadata": {
  "accelerator": "GPU",
  "colab": {
   "provenance": []
  },
  "gpuClass": "standard",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
